src/catalog/catalog_entry/schema_catalog_entry.cpp

```diff
@@ -31,12 +31,48 @@
 #include "duckdb/planner/parsed_data/bound_create_table_info.hpp"
 #include "duckdb/storage/data_table.hpp"
 #include "duckdb/common/field_writer.hpp"
+#include "duckdb/planner/constraints/bound_foreign_key_constraint.hpp"
+#include "duckdb/parser/constraints/foreign_key_constraint.hpp"
 
 #include <algorithm>
 #include <sstream>
 
 namespace duckdb {
 
+bool FindForeignKeyInformation(CatalogEntry *entry, string &pk_table, vector<string> &pk_columns,
+                               vector<idx_t> &pk_keys, vector<string> &fk_columns, vector<idx_t> &fk_keys,
+                               bool is_drop) {
+	bool is_found = false;
+	bool is_exception = false;
+	string exception_table;
+	if (entry->type == CatalogType::TABLE_ENTRY) {
+		auto *table_entry = (TableCatalogEntry *)entry;
+		for (idx_t i = 0; i < table_entry->constraints.size(); i++) {
+			auto &cond = table_entry->constraints[i];
+			if (cond->type == ConstraintType::FOREIGN_KEY) {
+				auto &foreign_key = (ForeignKeyConstraint &)*cond;
+				if (foreign_key.is_fk_table) {
+					pk_table = foreign_key.pk_table;
+					pk_columns = foreign_key.pk_columns;
+					pk_keys = foreign_key.pk_keys;
+					fk_columns = foreign_key.fk_columns;
+					fk_keys = foreign_key.fk_keys;
+					is_found = true;
+				} else if (is_drop) {
+					is_exception = true;
```

Comment:
Can we throw an exception here immediately, rather than setting variables?

Reply:
You are right. I will fix as you pointed.